sympy/ntheory/factor_.py

```diff
@@ -2371,3 +2371,73 @@ def is_amicable(m, n):
         return False
     a, b = map(lambda i: divisor_sigma(i), (m, n))
     return a == b == (m + n)
+
+
+def dra(n, b):
+    """
+    Returns the additive digital root of a natural number ``n`` in base ``b``
+    which is a single digit value obtained by an iterative process of summing
+    digits, on each iteration using the result from the previous iteration to
+    compute a digit sum.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import dra
+    >>> dra(3110, 12)
+    8
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Digital_root
+
+    """
+
+    num = abs(as_int(n))
+    b = (as_int(b))
+    if b <= 1:
+        raise ValueError("Base should be an integer greater than 1")
+
+    if num == 0:
+        return 0
+
+    return (1 + (num - 1) % (b - 1))
+
+
+def drm(n, b):
+    """
+    Returns the multiplicative digital root of a natural number ``n`` in a given
+    base ``b`` which is a single digit value obtained bu an iterative process of
+    multiplying digits, on each iteration using the result from the previous
+    iteration to compute the digit multiplication.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import drm
+    >>> drm(9876, 10)
+    0
+
+    >>> drm(49, 10)
+    8
+
+    References
+    ==========
+
+    .. [1] http://mathworld.wolfram.com/MultiplicativeDigitalRoot.html
+
+    """
+
+    n = abs(as_int(n))
```

Comment:
```suggestion
    n = abs(as_int(n))
    b = as_int(b)
```

Reply:
Will Flake checks allow the recasting of the variable or will you have to do `ni = abs(as_int(n))`?

Reply:
I am not sure about this.

Reply:
This is fine. What flake8 complains about is shadowing with a loop variable:
```python
from sympy.abc import x

def g():
    print(x)

def f():
    for x in range(2):
        pass
```
This gives
```console
$ flake8 tmp.py 
tmp.py:7:9: F402 import 'x' from line 1 shadowed by loop variable
```

Reply:
Thanks for the clarication.